client/snaptool.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type snaptoolOutput struct {
+	Stdout string
+	Stderr string
+}
+
+// RunSnaptool requests a snaptool run for the given context and arguments.
+func (client *Client) RunSnaptool(context string, args []string) (stdout string, stderr string, err error) {
```

Comment:
Done.

> Ah, as the result, let's please do (stdout, stderr []byte, err error), which is more typical for these values.

Note that []byte is json-encoded as a base64 string, so I had to convert shuffle between []byte and string in order to do this.